flink-staging/flink-streaming/flink-streaming-core/src/main/java/org/apache/flink/streaming/util/keys/KeySelectorUtil.java

```diff
@@ -152,10 +119,9 @@ public ArrayKeySelector(int... fields) {
 
 		@Override
 		public Tuple getKey(IN value) throws Exception {
-			key = (Tuple) tupleClasses[fields.length - 1].newInstance();
+			Tuple key = Tuple.getTupleClass(fields.length).newInstance();
 			for (int i = 0; i < fields.length; i++) {
-				int pos = fields[i];
```

Comment:
This was the bug... Wow. That's super critical... I am wondering whether this should be part of 0.10 milestone 1